### PR TITLE
UAVCAN Safety Button

### PR DESCRIPTION
**Pull request by dakejahl**
Brings this into the modern age

### CODE REVIEW

**Comment by dakejahl** [comment]
@AlexKlimaj Can you test this for me pretty please?

**Comment by AlexKlimaj** [comment]
@dagar The current safety button driver just publishes whether safety is enabled or disabled. The uavcan driver wants button pressed time. But dronecan has a 20000.SafetyState.uavcan and 20001.Button.uavcan message. Do we want the node to publish a raw button, or the safety state? Then we need to implement on the FC end.

**Comment by dakejahl** [comment]
Also the current uavcan impl for SafetyState is using the `actuator_armed` topic. Shouldn't it use `safety` instead?
https://github.com/PX4/PX4-Autopilot/blob/master/src/drivers/uavcan/safety_state.cpp#L69-L74

I will do the work just let me know what you'd like to see. Thanks!

File: src/drivers/uavcan/uavcan_params.c [hunk 1]
```diff
@@ -335,3 +335,15 @@ PARAM_DEFINE_INT32(UAVCAN_SUB_MAG, 1);
...
+ * subscription safety button
+ *
+ * Enable UAVCAN Safety Button.
+ *  ardupilot::indication::Button
+ *
+ * @boolean
+ * @reboot_required true
+ * @group UAVCAN
+ */
+PARAM_DEFINE_INT32(UAVCAN_SAFETY, 0);
```
Comment (dagar) [inline src/drivers/uavcan/uavcan_params.c:349]:
```suggestion
PARAM_DEFINE_INT32(UAVCAN_SUB_SAFETY, 0);
```

The convention I'm gradually working towards is everything being configurable via a `UAVCAN_SUB_*` or `UAVCAN_PUB_*` parameter.

**Pull request closed without merging**